src/main/java/reo/Parser.java

```diff
@@ -0,0 +1,135 @@
+                    String[] firstPart = parts[0].split(" ", 2);
+
+                    String name = firstPart[1];
+                    String deadline = parts[1].trim();
+
+                    Deadline deadlineToPush = new Deadline(name, false, deadline);
+                    tasks.addDeadline(deadlineToPush);
+                    storage.writeFile(deadlineToPush);
+                    return "I've added this deadline:\n" + deadlineToPush.toString() + "\n"
+                            + "Now, you have " + tasks.getSize() + " task(s) in your list.\n";
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    return "Please enter a valid task name and deadline.";
+                }
+            case EVENT:
+                try {
+                    String[] parts = input.split("/from", 2);
+                    String[] firstPart = parts[0].split(" ", 2);
+                    String[] secondPart = parts[1].split("/to", 2);
+
+                    String name = firstPart[1];
+                    String from = secondPart[0].trim();
+                    String to = secondPart[1].trim();
+
+                    Event eventToPush = new Event(name, false, to, from);
+                    tasks.addEvent(eventToPush);
+                    storage.writeFile(eventToPush);
+                    return "I've added this event:\n" + eventToPush.toString() + "\n"
+                            + "Now, you have " + tasks.getSize() + " task(s) in your list.\n";
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    return "Please enter a valid task name and to & from dates.";
+                }
+            case DELETE:
+                try {
+                    int index = Integer.valueOf(words[1]) - 1;
+                    // reo.Task toRemove = tasks.get(index);
+                    tasks.deleteTask(index);
+                    storage.removeLine(index + 1);
+                    Task toDelete = tasks.get(index);
+                    return "I've deleted this task:\n" + toDelete.toString() + "\n"
+                            + "Now, you have " + tasks.getSize() + " task(s) in your list.\n";
+                } catch (IndexOutOfBoundsException e) {
+                    return "Please enter a valid task number.\n";
+                }
+            case FIND:
+                try {
+                    String keyword = words[1];
+                    TaskList filtered = new TaskList(tasks.filter(keyword));
+                    return "Here are the matching tasks in your list:\n" + filtered.toString() + "\n";
+                } catch (IndexOutOfBoundsException e) {
+                    return "ERROR: Please enter a valid keyword to search for.";
+                }
+            case BYE:
+                return "Bye!";
+            case UNKNOWN:
+                return "ERROR: Please enter a valid command.\n";
+            default:
+                return "ERROR: Please enter a valid command.\n";
+            }
```

Comment:
For switch statements, case clauses should not have any indentation.

Method is also too long, consider using helper methods to handle each command.